src/types.ts

```diff
@@ -74,6 +79,23 @@ export interface IEmail {
   createdModSeq: number;
   updatedModSeq: number;
   deleted: Date | null;
+  receivedAt: Date;
```

Comment:
I wonder if this and other `Date` fields should even be present as `Date` fields. You can't pass Date via JSON, it will be number or string, which end user will then need to convert. They look nice in type, but I feel like this is the wrong type for it. Library is intended to work with raw data. And raw data type for `Date` is `string`/`number`.

Reply:
Yes (here in the specs https://jmap.io/spec-core.html#the-date-and-utcdate-data-types)

We could use string or a custom type
```typescript
export type IDate = string;
```

Reply:
I think this ^ would work.

Reply:
JMAP mandat String ( date-time format [@!RFC3339]): https://jmap.io/spec-core.html#the-date-and-utcdate-data-types